### PR TITLE
[Attribute] Fix hardcoded strings in attribute templates

### PR DESCRIPTION
**Pull request by Zales0123**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | https://github.com/Sylius/Sylius/issues/3984
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Sylius/Bundle/AttributeBundle/Resources/views/attributeTypes.html.twig [hunk 1]
```diff
@@ -8,7 +8,8 @@
             <div class="modal-body">
                 <div class="list-group">
                     {% for name, attributeType in attributeTypes %}
-                        <a href="{{ path('sylius_backend_product_attribute_create', { 'type': name }) }}" id="{{ name }}" class="list-group-item">
+                        {% set route = 'sylius_backend_'~subject~'_create' %}
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AttributeBundle/Resources/views/attributeTypes.html.twig]: I supposer the subject is `product`, so shouldn't it be "sylius_backend_'~subject~'_**attribute**_create"?

Comment (Zales0123) [inline src/Sylius/Bundle/AttributeBundle/Resources/views/attributeTypes.html.twig]: In fact, ``subject`` is "product_attribute"... perhaps variable name is not the best choice :smile:

File: src/Sylius/Bundle/AttributeBundle/Resources/views/attributeValueForms.html.twig [hunk 2]
```diff
@@ -14,9 +14,9 @@
             <div class="form-group">
                 {% set id = form.vars.label|replace(' ', '_')|lower %}
                 <label class="col-lg-2 control-label required" for="{{ id }}">{{ form.vars.label }}</label>
-                <input type="hidden" name="sylius_product[attributes][{{ count }}][attribute]" id="sylius_product_attributes_{{ count }}_attribute" value="{{ attributeId }}"/>
+                <input type="hidden" name="sylius_{{ subject }}[attributes][{{ count }}][attribute]" id="sylius_{{ subject }}_attributes_{{ count }}_attribute" value="{{ attributeId }}"/>
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AttributeBundle/Resources/views/attributeValueForms.html.twig]: The `[attributes]` aren't actually `[values]`?

File: src/Sylius/Bundle/AttributeBundle/Resources/views/attributeTypes.html.twig (hunk 1 above)
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AttributeBundle/Resources/views/attributeTypes.html.twig]: Then [see](https://github.com/Sylius/Sylius/pull/4014/files#r51257699) :)

File: src/Sylius/Bundle/AttributeBundle/Resources/views/attributeValueForms.html.twig (hunk 2 above)
Comment (Zales0123) [inline src/Sylius/Bundle/AttributeBundle/Resources/views/attributeValueForms.html.twig]: I think it was a talk about attributes collection name some time ago, but it has [not been fixed yet](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Component/Product/Model/Product.php#L56). Sure it should be, but I think it's not a case of this PR ;)

Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AttributeBundle/Resources/views/attributeValueForms.html.twig]: Ok :+1:

File: src/Sylius/Bundle/AttributeBundle/Resources/views/attributeTypes.html.twig [hunk 3]
```diff
@@ -8,7 +8,8 @@
             <div class="modal-body">
                 <div class="list-group">
                     {% for name, attributeType in attributeTypes %}
-                        <a href="{{ path('sylius_backend_product_attribute_create', { 'type': name }) }}" id="{{ name }}" class="list-group-item">
+                        {% set route = 'sylius_backend_'~subject~'_attribute_create' %}
```
Comment (peteward) [inline src/Sylius/Bundle/AttributeBundle/Resources/views/attributeTypes.html.twig]: what if you're application isn't sylius? I think you need to pass the metadata resource name here

File: src/Sylius/Bundle/AttributeBundle/Resources/views/attributeValueForms.html.twig (hunk 2 above)
Comment (peteward) [inline src/Sylius/Bundle/AttributeBundle/Resources/views/attributeValueForms.html.twig]: same here, need application name

File: src/Sylius/Bundle/AttributeBundle/Resources/views/attributeValueForms.html.twig [hunk 4]
```diff
@@ -33,7 +33,7 @@
         {% endfor %}
     {% elseif item.vars.name != '_token' %}
         {% set namePrefix = prefix|replace({'_': ']['}) %}
-        {% set dataName = 'sylius_product[attributes]['~count~namePrefix~']['~item.vars.name~']' %}
+        {% set dataName = 'sylius_'~subject~'[attributes]['~count~namePrefix~']['~item.vars.name~']' %}
```
Comment (peteward) [inline src/Sylius/Bundle/AttributeBundle/Resources/views/attributeValueForms.html.twig]: and anywhere else that's sylius_

**Comment by pjedrzejewski** [comment]
@Zales0123 You can pass ``$this->metadata`` and get resource name and application name from there.

File: src/Sylius/Bundle/AttributeBundle/Resources/views/attributeTypes.html.twig (hunk 3 above)
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AttributeBundle/Resources/views/attributeTypes.html.twig]: Hah :smile:  good point Peter :+1:

Comment (Zales0123) [inline src/Sylius/Bundle/AttributeBundle/Resources/views/attributeTypes.html.twig]: :+1: I think, I'll follow @pjedrzejewski advice and pass ``metadata`` object into these templates.

**Comment by michalmarcinkowski** [comment]
Looks good now :+1:

**Comment by pjedrzejewski** [comment]
Thank you Mateusz! :+1:

**Pull request merged**